### PR TITLE
internal/aws/awsutil: support per-component shared_credentials_file fields

### PR DESCRIPTION
**Pull request by jaylim-crl**
Previously, there was no way to use different AWS credentials for different
components. For instance, we can't use credential X for awsemfexporter, and
credential Y for awscloudwatchlogsexporter. This commit addresses that by
introducing a new per-component shared_credentials_file field, just like
role_arn, which will enable callers to configure specific AWS sessions for
different components.

### CODE REVIEW

**Comment by jaylim-crl** [comment]
Note that this merges into a `v0.89.0` base.

**Comment by alyshanjahani-crl** [comment]
Any reason why this is still in draft? lgtm

**Review by alyshanjahani-crl — APPROVED** [review]

File: internal/aws/awsutil/conn.go [hunk 1]
```diff
@@ -193,23 +193,33 @@ func ProxyServerTransport(logger *zap.Logger, config *AWSSessionSettings) (*http
...
+func (c *Conn) newAWSSession(
+	logger *zap.Logger, roleArn string, sharedCredentialsFile string, region string,
+) (*session.Session, error) {
 	var s *session.Session
 	var err error
-	if roleArn == "" {
-		s, err = GetDefaultSession(logger)
+	if sharedCredentialsFile != "" {
+		// profile defaults to "default" or the "AWS_PROFILE" environment variable.
+		sharedCreds := credentials.NewSharedCredentials(sharedCredentialsFile, "" /* profile */)
```
Comment (jaylim-crl) [inline internal/aws/awsutil/conn.go:203]: Note that we will default the profile to `default`, or the value defined in the `AWS_PROFILE` environment variable. If we decide we need different profiles in the future, we can customize this further.

**Review by jaylim-crl — COMMENTED** [review]
TFTR!

**Pull request merged**